src/components/NewTodoForm.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from 'react';
+import { useAppDispatch } from '../hooks';
+import { format, addDays } from 'date-fns';
+import { addTodo } from '../store/todoSlice';
+import { TextField, Paper, Button, Grid } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import Modal from './Modal';
+import { DATE_FORMAT } from '../constants';
+import '../style.css'
+
+const NewTodoForm: React.FC = () => {
+    const [text, setText] = useState('');
+    const [showModal, setShowModal] = useState(false);
+    const dispatch = useAppDispatch();
+
+    const handleCloseModule = () => {
```

Comment:
*handleCloseModAl, 
Why did you create function only for closing ? When need to open modal - you use onClick={() => {
                                    setShowModal(true)
                                }}
You can create one function and pass true/false, or do setShowModal(!showModal) in that function